programs/account-compression/src/instructions/append_leaves.rs

```diff
@@ -81,20 +82,30 @@ fn process_batch<'a, 'c: 'info, 'info>(
             let batch_size = end - start;
             leaves_processed += batch_size;
 
-            let merkle_tree_account =
-                AccountLoader::<StateMerkleTreeAccount>::try_from(merkle_tree_acc_info)
-                    .map_err(ProgramError::from)?;
+            let rollover_fee = {
+                let merkle_tree_account =
+                    AccountLoader::<StateMerkleTreeAccount>::try_from(merkle_tree_acc_info)
+                        .map_err(ProgramError::from)?;
 
-            let mut merkle_tree_account = merkle_tree_account.load_mut()?;
-            let rollover_fee =
-                merkle_tree_account.metadata.rollover_metadata.rollover_fee * batch_size as u64;
+                let rollover_fee = {
```

Comment:
why nest this?

Reply:
Because the loaded Merkle tree (with Anchor's `load()`/`load_mut()`) needs to be dropped. Otherwise the program will be aborted with a runtime error that we are borrowing an account twice.

An another option to achieve this would be an explicit drop (like `mekle_tree.drop();`), but I usually prefer freeing memory with scopes rather than explicit drops.